ambry-rest/src/test/java/com.github.ambry.rest/NettyRequestTest.java

```diff
@@ -758,19 +983,53 @@ private void doContentAddAndReadTest(String digestAlgorithm, ByteBuffer content,
    * @return the whole content as a {@link ByteBuffer} - serves as a source of truth.
    */
   private ByteBuffer generateContent(List<HttpContent> httpContents) {
-    int individualPartSize = GENERATED_CONTENT_SIZE / GENERATED_CONTENT_PART_COUNT;
+    return generateContent(httpContents, false);
+  }
+
+  /**
+   * Generates random content and fills it up (in parts) in {@code httpContents}.
+   * @param httpContents the {@link List<HttpContent>} that will contain all the content in parts.
+   * @param useCopyForcingByteBuf if {@code true}, uses {@link CopyForcingByteBuf} instead of the default
+   *                              {@link ByteBuf}.
+   * @return the whole content as a {@link ByteBuffer} - serves as a source of truth.
+   */
+  private ByteBuffer generateContent(List<HttpContent> httpContents, boolean useCopyForcingByteBuf) {
     byte[] contentBytes = RestTestUtils.getRandomBytes(GENERATED_CONTENT_SIZE);
-    for (int addedContentCount = 0; addedContentCount < GENERATED_CONTENT_PART_COUNT - 1; addedContentCount++) {
-      HttpContent httpContent = new DefaultHttpContent(
-          Unpooled.wrappedBuffer(contentBytes, addedContentCount * individualPartSize, individualPartSize));
-      httpContents.add(httpContent);
-    }
-    httpContents.add(new DefaultLastHttpContent(Unpooled
-        .wrappedBuffer(contentBytes, (GENERATED_CONTENT_PART_COUNT - 1) * individualPartSize, individualPartSize)));
+    splitContent(contentBytes, GENERATED_CONTENT_PART_COUNT, httpContents, useCopyForcingByteBuf);
     return ByteBuffer.wrap(contentBytes);
   }
 
   /**
+   * Splits the given {@code contentBytes} into {@code numChunks} chunks and stores them in {@code httpContents}.
+   * @param contentBytes the content that needs to be split.
+   * @param numChunks the number of chunks to split {@code contentBytes} into.
+   * @param httpContents the {@link List<HttpContent>} that will contain all the content in parts.
+   * @param useCopyForcingByteBuf if {@code true}, uses {@link CopyForcingByteBuf} instead of the default
+   *                              {@link ByteBuf}.
+   */
+  private void splitContent(byte[] contentBytes, int numChunks, List<HttpContent> httpContents,
```

Comment:
I'm assuming you mean full request i.e `FullHttpRequest` that contains both the request and content. That test is done as a part of `NettyMessageProcessorTest::rawBytesPostTest()` where it belongs.